Parser/pegen/pegen.c

```diff
@@ -2074,3 +2074,42 @@ _PyPegen_make_module(Parser *p, asdl_seq *a) {
     }
     return Module(a, type_ignores, p->arena);
 }
+
+// Error reporting helpers
+
+expr_ty
+_PyPegen_get_invalid_target(expr_ty e)
+{
+    if (e == NULL) {
+        return NULL;
+    }
+    switch (e->kind) {
+        case List_kind: {
+            Py_ssize_t len = asdl_seq_LEN(e->v.List.elts);
+            for (Py_ssize_t i = 0; i < len; i++) {
+                expr_ty other = asdl_seq_GET(e->v.List.elts, i);
+                if (_PyPegen_get_invalid_target(other)) {
+                    return other;
+                }
+            }
+            return NULL;
+        }
+        case Tuple_kind: {
+            Py_ssize_t len = asdl_seq_LEN(e->v.Tuple.elts);
+            for (Py_ssize_t i = 0; i < len; i++) {
+                expr_ty other = asdl_seq_GET(e->v.Tuple.elts, i);
+                expr_ty child = _PyPegen_get_invalid_target(other);
+                if (child != NULL) {
+                    return child;
+                }
+            }
+            return NULL;
+        }
+        case Starred_kind:
+            return _PyPegen_get_invalid_target(e->v.Starred.value);
+        case Name_kind:
+            return NULL;
```

Comment:
I guess this should also include `Attribute`s and `Subscript`s as well?

Reply:
Yup, I added some extra tests for this